SonarQube.Client.Tests/Models/ServerSentEvents/SSEStreamWriterTests.cs

```diff
@@ -0,0 +1,146 @@
+﻿/*
+ * SonarQube Client
+ * Copyright (C) 2016-2023 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System.IO;
+using System.Text;
+using System.Threading;
+using System.Threading.Channels;
+using System.Threading.Tasks;
+using FluentAssertions;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using Moq;
+using SonarQube.Client.Models.ServerSentEvents;
+using SonarQube.Client.Models.ServerSentEvents.ServerContract;
+
+namespace SonarQube.Client.Tests.Models.ServerSentEvents
+{
+    [TestClass]
+    public class SSEStreamWriterTests
+    {
+        [TestMethod, Timeout(10000)]
+        public async Task BeginListening_StreamEnds_TaskFinishes()
+        {
+            var streamReader = CreateStreamReader(content: "some data\nanother data\n");
+            var channel = CreateChannel();
+
+            var testSubject = CreateTestSubject(streamReader, channel);
+
+            await testSubject.BeginListening();
+
+            streamReader.EndOfStream.Should().BeTrue();
+            channel.Reader.Count.Should().Be(0);
+        }
+
+        [TestMethod, Timeout(10000)]
```

Comment:
why Timeout?

Reply:
In case there is a bug, and the async hangs. So this way the test will terminate